components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/CORSRequestHandler.java

```diff
@@ -16,69 +16,191 @@
 import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
 
+import java.util.Arrays;
+import java.util.List;
 import java.util.Map;
 
-/**
- * This handler functions as an enabler for the Store API Console
- * Sets the Access-Control-Allow-Origin header
- *
- */
 public class CORSRequestHandler extends AbstractHandler implements ManagedLifecycle {
 
-    private static final Log log = LogFactory.getLog(CORSRequestHandler.class);
-
-    public void init(SynapseEnvironment synapseEnvironment) {
-        log.debug("Initializing API authentication handler instance");
-    }
-
-    public void destroy() {
-        log.debug("Destroying API authentication handler instance");
-    }
-
-    public boolean handleRequest(MessageContext messageContext) {
-
-        if (Utils.isCORSEnabled()) {
-	    	/* For CORS support adding required headers to the response */
-            org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
-                    getAxis2MessageContext();
-            Map<String, String> headers = (Map) axis2MC.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
-            String requestOrigin = (String) headers.get("Origin");
-
-            headers.put(APIConstants.CORSHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, Utils.getAllowedOrigin(requestOrigin));
-            headers.put(APIConstants.CORSHeaders.ACCESS_CONTROL_ALLOW_METHODS, Utils.getAllowedMethods());
-            headers.put(APIConstants.CORSHeaders.ACCESS_CONTROL_ALLOW_HEADERS, Utils.getAllowedHeaders());
-            axis2MC.setProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS, headers);
-        }
-
-        return true;
-    }
-
-    public boolean handleResponse(MessageContext messageContext) {
-
-    	if (Utils.isCORSEnabled()) {
-	    	/* For CORS support adding required headers to the response */
-	    	org.apache.axis2.context.MessageContext axis2MC = ((Axis2MessageContext) messageContext).
-	                getAxis2MessageContext();
-	    	Map<String, String> headers = (Map) axis2MC.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
-            String requestOrigin = (String) headers.get("Origin");
-	    		    	    	
-	    	headers.put(APIConstants.CORSHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, Utils.getAllowedOrigin(requestOrigin));
-	        headers.put(APIConstants.CORSHeaders.ACCESS_CONTROL_ALLOW_METHODS, Utils.getAllowedMethods());
-	        headers.put(APIConstants.CORSHeaders.ACCESS_CONTROL_ALLOW_HEADERS, Utils.getAllowedHeaders());
-	        axis2MC.setProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS, headers);
-    	}
- 
-        return true;
-    }
+	private static final Log log = LogFactory.getLog(CORSRequestHandler.class);
+	private String inline;
+	private String allowHeaders;
+	private List<String> allowedOrigins;
+	private boolean headerStatus;
```

Comment:
Variable name doesn't sound right. What does this boolean do? Name it accordingly.